### ISSUE TITLE
Unterminated #endif in superlu_config.fh with CMake

### ISSUE THREAD

**Issue by ArmstrongJ** [xiaoyeli/superlu#126]
When I build with CMake and _-Denable_fortran=YES_, the build will fail with

```superlu\superlu-6.0.1\FORTRAN\superlu_config.fh:2:0:

    2 | #ifndef SUPERLU_CONFIG_H
      |
Error: unterminated #ifndef
FORTRAN\CMakeFiles\dfexm.dir\build.make:88: recipe for target 'FORTRAN/CMakeFiles/dfexm.dir/hbcode1.F90.obj' failed
```

The error stems from the steps to regenerate _superlu_config.fh_.  Specifically, the CMake-generate sed command in the Makefile:

```
cd /d C:\superlu\build64\FORTRAN && sed '/^\// d' < superlu_config.h > temp.fh
```

Testing the sed command, you'll see that it specifically strips the line:

```
#endif /* SUPERLU_CONFIG_H */
```

out of the file entirely since it has a C comment on it.  The problem originates from FORTRAN/CMakeLists.txt:96 for reference.

Also, somewhat related, but generating a file in the source tree seems highly problematic as well.  I would argue that _superlu_config.fh_ should really be generated in the build directory (probably in the _FORTRAN_ subdirectory).

**Comment by ArmstrongJ** [comment]
Pull Request #127 should fix this issue as well as eliminate the generation of _superlu_config.fh_ within the source tree when using CMake.

**Comment by gruenich** [comment]
Fortran works for me on Linux with CMake. The header contains the `#endif`. Probably a Windows issues. That's why it got unnoticed.  
Anyway, should be fixed!

**Comment by xiaoyeli** [comment]
This should be resolved from #135
